### PR TITLE
Bumped go.mod version in tools

### PR DESCRIPTION
**Pull request by anshuman-agarwala**
**What this PR does / why we need it**: Fixes the lint failure here: https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/2071
This is happening because dependabot does not update the go.mod version automatically and our CI checks that running `go mod tidy` does not make any further changes to the go.mod file. The version change is happening because go-lint depends on version 1.22.1 which is due to this dependency in go-lint: honnef.co/go/tools

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: hack/tools/go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/cluster-api-provider-ibmcloud/hack/tools
 
-go 1.22.0
+go 1.22.10
```
Comment (mkumatag) [inline hack/tools/go.mod:3]: right now we are using `1.22.9` version, we need to limit to that or bump the version for the project before this change

cc @Amulyam24 @Prajyot-Parab

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: hack/tools/go.mod (hunk 1 above)
Comment (anshuman-agarwala) [reply hack/tools/go.mod]: Changed version from `1.22.10` to `1.22.9` to be in line with our go version.

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**